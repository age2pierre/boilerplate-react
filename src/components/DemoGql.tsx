import gql from 'graphql-tag'
import React from 'react'
import { useQuery } from 'urql'
import type { ValuesType } from 'utility-types'

import type {
  DemoGetFilms,
  DemoGetFilmsVariables,
} from './__generated__/demo-get-films'

export type Film = ValuesType<
  NonNullable<NonNullable<DemoGetFilms['allFilms']>['films']>
>

export const FilmList = (props: { films: Film[] }) => {
  return (
    <ul>
      {props.films.map(film => (
        <li key={film?.id}>{film?.title ?? 'N/A'}</li>
      ))}
    </ul>
  )
}

export const DemoPage = () => {
  const [{ fetching, data, error }] = useQuery<
    DemoGetFilms,
    DemoGetFilmsVariables
  >({
    query: gql`
      query DemoGetFilms {
        allFilms {
          totalCount
          films {
            id
            title
            created
          }
        }
      }
    `,
  })

  const paragraph: string = fetching
    ? 'Please wait...'
    : error
    ? `errors => ${error.message}`
    : `Found ${data?.allFilms?.totalCount ?? 0} movies`

  return (
    <>
      <h1>SWAPI Example</h1>
      <p>{paragraph}</p>
      <FilmList films={data?.allFilms?.films ?? []} />
    </>
  )
}

export default DemoPage
