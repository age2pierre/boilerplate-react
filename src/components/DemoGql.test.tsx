import React from 'react'
import { create } from 'react-test-renderer'

import { FilmList } from './DemoGql'

it('FilmList renders correctly', () => {
  const comp = (
    <FilmList films={[{ id: '0007', title: 'James Bond', created: null }]} />
  )
  const tree = create(comp).toJSON()
  expect(tree).toMatchSnapshot()
})
