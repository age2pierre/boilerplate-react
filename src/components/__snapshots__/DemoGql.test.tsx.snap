// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList renders correctly 1`] = `
<ul>
  <li>
    James Bond
  </li>
</ul>
`;
